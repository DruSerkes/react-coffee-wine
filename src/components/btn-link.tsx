import * as React from 'react';



const BtnLink = ()