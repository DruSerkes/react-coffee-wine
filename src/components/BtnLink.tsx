import * as React from 'react';
import { BtnLinkProps } from '../types/btn-link';


export const BtnLink = ({ text, color, link }: BtnLinkProps) => {
    return (
        <button style={{ backgroundColor: color }} className="BtnLink">
            <a href={link}>
                {text}
            </a>
        </button>
    )
}