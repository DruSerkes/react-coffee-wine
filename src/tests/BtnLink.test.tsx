import * as React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { BtnLink } from '../components/BtnLink'


describe('BtnLink Tests', () => {
    it('should pass smoke/snapshot test', () => {
        const { asFragment } = render(<BtnLink text="Test" color="#2F2F2F" link="https://test.com/test" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render text "test"', () => {
        render(<BtnLink text="Test" color="#2F2F2F" link="https://test.com/test" />);
        const testButton = screen.getByText('Test');
        expect(testButton).toBeInTheDocument();
    })

    it('should redirect to google.com', () => {
        render(<BtnLink text="Test" color="#2F2F2F" link="https://google.com/" />);
        const testButton = screen.getByText('Test');
        fireEvent.click(testButton);
        waitFor(() => {
            expect(window.location).toBe('https://google.com/');
        })

    })
})