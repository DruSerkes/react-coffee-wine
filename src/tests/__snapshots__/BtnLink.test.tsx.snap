// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnLink Tests should pass smoke/snapshot test 1`] = `
<DocumentFragment>
  <button
    class="BtnLink"
    style="background-color: rgb(47, 47, 47);"
  >
    <a
      href="https://test.com/test"
    >
      Test
    </a>
  </button>
</DocumentFragment>
`;
